src/libdrakvuf/libdrakvuf.h

```diff
@@ -274,6 +302,32 @@ char *drakvuf_get_current_process_name(drakvuf_t drakvuf,
                                        uint64_t vcpu_id,
                                        x86_registers_t *regs);
 
+
+bool drakvuf_get_current_thread_id( drakvuf_t drakvuf, 
+                                    uint64_t vcpu_id, 
+                                    x86_registers_t *regs,
+                                    uint32_t *thread_id );
+
+bool drakvuf_get_previous_mode( drakvuf_t drakvuf, 
```

Comment:
Oops!! Missed your previous post, my apologies... :)

About adding the link: that's a good point. :)

Regarding the thread address: according to my experience most use comes from inside hooks (SSDT, Shadow) to know from where it's called, anyway let me take a look a few minutes...